app/controllers/api/consignments_controller.rb

```diff
@@ -5,22 +5,42 @@ class ConsignmentsController < RestController
     before_action :require_authentication
 
     def update_price
-      artworks_price = fetch_sale_artworks_with_price(params[:sale_id])
+      update_sale_info(params[:sale_id])
 
-      PartnerSubmissionService.update_price(artworks_price)
       render json: { result: 'ok' }, status: :created
     end
 
     private
 
-    def fetch_sale_artworks_with_price(sale_id)
+    def update_sale_info(sale_id)
+      sale = Gravity.client.sale(id: sale_id)._get
       sale_artworks =
         Gravity.client.sale_artworks(sale_id: sale_id).sale_artworks
 
       sale_artworks.map do |sale_artwork|
-        artwork_id = sale_artwork.artwork.id
+        artwork = sale_artwork.artwork
+        submission = Submission.find_by(source_artwork_id: artwork.id)
+        next unless submission
+
+        submission.assign_attributes(
+          title: artwork.title,
+          medium: artwork.medium
+        )
+        submission.save!
+
+        consignment = submission.consigned_partner_submission
+        next unless consignment
+
         price = sale_artwork.highest_bid.try(:amount_cents)
-        [artwork_id: artwork_id, price: price]
+        state = price ? 'sold' : 'bought in'
+        consignment.assign_attributes(
+          sale_price_cents: price || consignment.sale_price_cents,
+          sale_lot_number: sale_artwork.lot_number,
+          sale_date: sale.end_date,
+          state: state,
+          sale_name: sale.name
+        )
+        consignment.save!
```

Comment:
**suggestion:** We should update this logic in [`PartnerSubmissionService#update_price`](https://github.com/artsy/convection/blob/9918182bc6950f0442a1390d47128864e76beb26/app/services/partner_submission_service.rb#L108-L118) and call that method from here.